app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -4134,19 +4147,62 @@ public AppCompatActivity getParentActivity() {
         return (AppCompatActivity) ((ViewGroup) binding.getRoot().getParent()).getContext();
     }
 
-    private void useVideoSource(final boolean video) {
-        if (playQueue == null || isAudioOnly == !video || audioPlayerSelected()) {
+    private void useVideoSource(final boolean videoEnabled) {
+        if (playQueue == null || isAudioOnly == !videoEnabled || audioPlayerSelected()) {
             return;
         }
 
-        isAudioOnly = !video;
+        isAudioOnly = !videoEnabled;
         // When a user returns from background controls could be hidden
         // but systemUI will be shown 100%. Hide it
         if (!isAudioOnly && !isControlsVisible()) {
             hideSystemUIIfNeeded();
         }
+
+        final int videoRenderIndex = getVideoRendererIndex();
+
+        // We can safely assume that currentMetadata is not null (otherwise this method isn't
+        // called) so we can use the requireNonNull method of the Objects class.
+        final StreamInfo info = Objects.requireNonNull(currentMetadata).getMetadata();
+
+        /* For video streams: we don't want to stream in background the video stream so if the
+        video stream played is not a video-only stream and if there is an audio stream available,
+        play this audio stream in background by reloading the play queue manager.
+        Otherwise the video renderer will be just disabled (because there is no
+        other stream for it to play the audio): if the video stream is video-only, only the audio
+        stream will be fetched and the video stream will be fetched again when the user return to a
+        video player.
+
+        For audio streams: nothing is done, it's not needed to reload the player with the same
+        audio stream.
+
+        In the case where we don't know the index of the video renderer, the play queue manager
+        is also reloaded. */
+
+        final StreamType streamType = info.getStreamType();
+
+        final boolean isVideoStreamTypeAndIsVideoOnlyStreamOrNoAudioStreamsAvailable =
+                (streamType == StreamType.VIDEO_STREAM || streamType == StreamType.LIVE_STREAM)
+                        && (videoResolver.wasLastResolvedVideoAndAudioSeparated().orElse(false)
+                            || isNullOrEmpty(info.getAudioStreams()));
```

Comment:
Could you explain why `|| isNullOrEmpty(info.getAudioStreams()`?

Reply:
`isNullOrEmpty(info.getAudioStreams()` refers to the `NoAudioStreamsAvailable` part of the boolean name.

Why I did this? Because it's not needed to reload the play queue manager to try to get audio streams if there are no audio streams (like on most of old PeerTube videos): it will reload the same video stream (which is not video-only and contains audio) in this case, which is not needed.

That's the reason why I said `Otherwise the video renderer will be just disabled (because there is no other stream for it to play the audio)` in my comment above this line.

Reply:
Oh ok, that makes sense